private/bufpkg/bufcheck/bufbreaking/bufbreakingconfig/bufbreakingconfig.go

```diff
@@ -21,95 +21,92 @@ import (
 	"github.com/bufbuild/buf/private/bufpkg/bufcheck/internal"
 )
 
-// Rule is a rule.
-type Rule interface {
-	bufcheck.Rule
-
-	// InternalRule returns the internal Rule.
-	InternalRule() *internal.Rule
-}
+const (
+	v1Beta1Version = "v1beta1"
+	v1Version      = "v1"
+)
 
-// Config is the check config.
+// Config is the breaking check config.
 type Config struct {
-	// Rules are the rules to run.
-	//
-	// Rules will be sorted by first categories, then id when Configs are
-	// created from this package, i.e. created wth ConfigBuilder.NewConfig.
-	Rules                  []Rule
-	IgnoreIDToRootPaths    map[string]map[string]struct{}
-	IgnoreRootPaths        map[string]struct{}
+	Use    []string
+	Except []string
```

Comment:
Note that this means we'll generate a new digest when a user changes their configured rules/categories even if it won't have an effect on the resulting `[]bufcheck.Rule`.

I actually think this is good since it means we're capturing changes in the users' `buf.yaml` configuration. Plus, it will prevent a hard-to-debug edge case when a user thinks they've made a change to their `lint` configuration, but their push gets rejected by the server since it looks like the same content (because the rules get unrolled to the same set as the previous commit).

Reply:
I believe this is very reasonable. I mean, if we really wanted to capture the rules as part of the digest, we can always build the rules during the digest and use that, but I think encoding the config structure in this manner is also very reasonable.